diff --git a/lib/rucio/transfertool/fts3.py b/lib/rucio/transfertool/fts3.py
index 7aef2ac82..343f58c07 100644
--- a/lib/rucio/transfertool/fts3.py
+++ b/lib/rucio/transfertool/fts3.py
@@ -973,9 +973,21 @@ class FTS3Transfertool(Transfertool):
             if all(_use_tokens(t) for t in sub_path):
                 logger(logging.DEBUG, 'OAuth2/OIDC available for transfer {}'.format([str(hop) for hop in sub_path]))
                 oidc_support = True
-            return sub_path, TransferToolBuilder(cls, external_host=fts_hosts[0], oidc_support=oidc_support, vo=vo)
+            is_tape_source = any(
+                t.src.rse.is_tape_or_staging_required()
+                for t in sub_path
+            )
+            external_host = (
+                'https://fts3-cms.cern.ch:8446'
+                if fts_hosts[0] == 'https://fts4-cms.cern.ch:8446' and is_tape_source
+                else fts_hosts[0]
+            )
+            if (external_host == 'https://fts4-cms.cern.ch:8446'
+                    and config_get_bool('experimental', 'override-fts4', default=False)):
+                external_host = 'https://fts3-cms.cern.ch:8446'
+            return sub_path, TransferToolBuilder(cls, external_host=external_host, oidc_support=oidc_support, vo=vo)
         else:
             return [], None
 
     def group_into_submit_jobs(self, transfer_paths: "list[list[DirectTransfer]]") -> list[dict[str, Any]]:
         jobs = bulk_group_transfers(